04_numeronym/shahriys/main_test.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestMain(t *testing.T) {
+	var bufnum bytes.Buffer
+	outnum = &bufnum
+
+	main()
+	expected := `[a11y K8s abc]`
+	actual := bufnum.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()" + expected + " " + actual)
+	}
+
+}
+func TestNumeronym(t *testing.T) {
+
+	type test struct {
+		expected []string
+		actual   []string
+	}
+
+	tests := []test{
+		{expected: []string{"a2c"}, actual: numeronyms("abac")},
+		{expected: []string{}, actual: numeronyms("")},
```

Comment:
I'm surprised this test passes. An empty string `""` should have numeronym `""`, not nothing. I would expect this to return `[]string{""}`